### PR TITLE
(CDAP-16008) Added pre-job validation for kerberos information

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillPreparer.java [hunk 1]
```diff
@@ -147,6 +155,30 @@ protected void launch(TwillRuntimeSpecification twillRuntimeSpec, RuntimeSpecifi
     }
   }
 
+  private void validateKerberos(@Nullable String principal, @Nullable String keytab, SSHSession session) {
+    if (principal == null || keytab == null) {
```
Comment (CuriousVini) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillPreparer.java:159]: should we check for null or empty?

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillPreparer.java [hunk 2]
```diff
@@ -147,6 +155,30 @@ protected void launch(TwillRuntimeSpecification twillRuntimeSpec, RuntimeSpecifi
...
+      return;
+    }
+
+    String klistPath;
+    try {
+      klistPath = session.executeAndWait("which klist").trim();
+    } catch (IOException e) {
+      LOG.warn("Failed to locate klist command for Kerberos validation. " +
+                 "If the Kerberos principal and keytab are mis-configured, program execution will fail.", e);
```
Comment (CuriousVini) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillPreparer.java:168]: should this message include error message from `e`?

**Review by yaojiefeng — APPROVED** [review]
One question, LGTM

**Review by CuriousVini — COMMENTED** [review]
minor questions. LGTM 👍🏼

**Review by CuriousVini — APPROVED** [review]

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillPreparer.java [hunk 3]
```diff
@@ -147,6 +155,30 @@ protected void launch(TwillRuntimeSpecification twillRuntimeSpec, RuntimeSpecifi
...
+  private void validateKerberos(@Nullable String principal, @Nullable String keytab, SSHSession session) {
+    if (principal == null || keytab == null) {
+      return;
+    }
+
+    String klistPath;
+    try {
+      klistPath = session.executeAndWait("which klist").trim();
+    } catch (IOException e) {
+      LOG.warn("Failed to locate klist command for Kerberos validation. " +
```
Comment (yaojiefeng) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillPreparer.java:167]: Why do we just log and return here?

**Pull request merged**